### ISSUE TITLE
jax vs numpy array inconsistent behavior in matplotlib histograms

### ISSUE THREAD

**Issue by henrypinkard** [google/jax#9007]
There's inconsistent behavior when using jax.numpy arrays vs numpy arrays when plotting histograms in matplotlib. I'm guessing this is a result of some differing property of the two objects thats not immediately obvious to me. They have the same shape. Its easy to fix by just converting to a regular numpy array (as below), but I thought I would report in case this is indicative of something that might cause problems in other situations

```python
import matplotlib.pyplot as plt
import jax.numpy as np
import numpy as onp

a = onp.random.rand(100)

plt.figure()
_ = plt.hist(np.array(a))

plt.figure()
_ = plt.hist(onp.array(a))
```

gives:
![Screen Shot 2021-12-17 at 9 43 40 AM](https://user-images.githubusercontent.com/7969470/146586234-29ea6f14-11d0-4085-9582-695391900812.png)

**Comment by jakevdp** [comment]
This looks like a duplicate of #8899 - unfortunately, it’s not something that can be fixed within JAX, but rather an issue with how matplotlib treats non-numpy inputs. The best workaround is to convert your JAX arrays to numpy arrays before passing them to matplotlib.

**Comment by jakevdp** [comment]
This is fixed now in matplotlib: https://github.com/matplotlib/matplotlib/pull/22018

**Comment by xinglong-li** [comment]
It seems this issue is fixed in matplotlib-3.5.2, but exist in earlier versions of matplotlib.
